core-java-modules/core-java-concurrency-collections-2/src/main/java/com/baeldung/concurrent/lock/ConcurrentAccessBenchmark.java

```diff
@@ -0,0 +1,52 @@
+package com.baeldung.concurrent.lock;
+
+import java.util.HashMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+@State(Scope.Thread)
+@Fork(value = 2)
+@Warmup(iterations = 0)
+public class ConcurrentAccessBenchmark {
+    ConcurrentAccessExperiment accessMyMap;
+    static final int SLOTS = 4;
+    static final int THREADS = 10000;
+    static final int BUCKETS = Runtime.getRuntime().availableProcessors() * SLOTS;
+
+    @Benchmark
+    @BenchmarkMode(Mode.Throughput)
+    @OutputTimeUnit(TimeUnit.MILLISECONDS)
+    public void singleLockHashMap() throws InterruptedException {
+        (new SingleLock()).doWork(new HashMap<String,String>(), THREADS, SLOTS);
```

Comment:
I don't understand why you are re-instantiated the object each time. Won't that play into your results?

It seems like what you want to do is just reuse one already instantiated at the field level. That way, your benchmark is only about the work being done on the map and not about that plus the cost of instantiating the worker object.

Reply:
I'm sorry but I don't get the point. Instantiating the object is not such a big cost, also it appears in every test. If we want to instantiate the maps before, we should also remove the "put" phase inside the test.

Reply:
To be clear, I'm only referring to `new SingleLock()` and by association `new StripedLock()`.

The reason we should remove the instantiation is that it is unnecessary. What are you gaining from instantiating it here?

Reply:
Ok